cabal-install/Distribution/Client/Security/HTTP.hs

```diff
@@ -0,0 +1,307 @@
+{-# LANGUAGE CPP #-}
+{-# LANGUAGE DeriveDataTypeable #-}
+{-# LANGUAGE FlexibleContexts #-}
+{-# LANGUAGE RankNTypes #-}
+{-# LANGUAGE RecordWildCards #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+{-# LANGUAGE StandaloneDeriving #-}
+-- | Implementation of 'HttpClient' using the HTTP package
+--
+-- This is adapted from the hackage-security-http package. Most important
+-- difference is that it includes the hack to deal with proxies that
+-- transparently decompress gzip streams without informing the client.
+module Distribution.Client.Security.HTTP (
+    withClient
+    -- ** Additional operations
+  , setOutHandler
+  , setErrHandler
+  , setProxy
+  , request
```

Comment:
This is a direct copy of the `hackage-security-HTTP` library, that's why. I'll strip it down (though if we merge it with the transport concept already existent in `cabal-install`, this will change anyway).